src/persist-types/src/arrow.rs

```diff
@@ -420,6 +420,49 @@ pub struct ArrayIdx<'a> {
     pub array: &'a ArrayOrd,
 }
 
+#[inline]
+fn list_range<'a>(
+    offsets: &OffsetBuffer<i32>,
+    values: &'a ArrayOrd,
+    idx: usize,
+) -> impl Iterator<Item = ArrayIdx<'a>> {
+    let offsets = offsets.inner();
+    let from = offsets[idx].as_usize();
+    let to = offsets[idx + 1].as_usize();
+    (from..to).map(|i| values.at(i))
+}
+
+impl<'a> ArrayIdx<'a> {
+    /// Returns the rough amount of space required for this entry in bytes.
+    /// (Not counting nulls, dictionary encoding, or other space optimizations.)
+    pub fn data_bytes(&self) -> usize {
```

Comment:
Thinking of changing this to `goodbytes` and counting only data bytes and not metadata like line lengths. (Makes it a worse approximation of the actual memory size, but closer to the number we use for other similar decisions like when to keep a part inline or write it out.)

Reply:
Seems reasonable to me!